command-line/lib/Command/Script.hs

```diff
@@ -0,0 +1,162 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Command.Script
+    ( Cmd
+    , mod
+    , run
+
+    -- * Internal
+    , requireSignatureOfParser
+    , requireAllOfParser
+    , requireAnyOfParser
+    , requireAtLeastOfParser
+    , scriptParser
+    ) where
+
+import Cardano.Multisig
+    ( Script (..), keyHashFromBytes, toScriptHash, validateScript )
+import Codec.Binary.Bech32.TH
+    ( humanReadablePart )
+import Codec.Binary.Encoding
+    ( fromBase16 )
+import Data.Char
+    ( isDigit, isLetter )
+import Data.Word
+    ( Word8 )
+import Options.Applicative
+    ( CommandFields, Mod, command, footerDoc, header, helper, info, progDesc )
+import Options.Applicative.Encoding
+    ( Encoding, encodingOpt )
+import Options.Applicative.Help.Pretty
+    ( bold, indent, string, vsep )
+import Prelude hiding
+    ( mod )
+import System.IO
+    ( stdin, stdout )
+import System.IO.Extra
+    ( hPutBytes, noNewline, progName )
+import Text.ParserCombinators.ReadP
+    ( ReadP, readP_to_S, (<++) )
+
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Text.ParserCombinators.ReadP as P
+
+
+newtype Cmd = Cmd
+    { encoding :: Encoding
+    } deriving (Show)
+
+mod :: (Cmd -> parent) -> Mod CommandFields parent
+mod liftCmd = command "scripthash" $
+    info (helper <*> fmap liftCmd parser) $ mempty
+        <> progDesc "Create a script hash"
+        <> header "Create a script hash \
+            \that can be used in stake or payment address."
+        <> footerDoc (Just $ vsep
+            [ string "The script is read from stdin."
```

Comment:
Wouldn't it be better to have the script as an argument here? There's no other command returning a script (and I can't see a use case for one), so piping scripts through stdin sounds weird.

Reply:
well, when checking today I did : 
```
$ echo "all [$(cat verKeyHash1),$(cat verKeyHash2)]" \
cardano-address scripthash
```
;-)

Reply:
Yes, what's the added value compared to:

```console
$ cardano-address scripthash "all [$(cat verKeyHash1),$(cat verKeyHash2)]" 
```

:) ?

Reply:
as I said, I am not stating this is not weird ;-)